Type 'Float64Array' is missing the following properties from type 'number[]': pop, push, concat, shift, and 6 more.

```bash
Run yarn next build
yarn run v1.22.22
$ /home/runner/work/sc4pe_pre2/sc4pe_pre2/node_modules/.bin/next build
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

  ▲ Next.js 14.2.5

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types ...

./src/components/RandomMesh.tsx
28:18  Warning: Assignments to the 'renderer' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
Failed to compile.

./src/components/RandomMesh.tsx:89:25
Type error: Conversion of type 'TypedArray' to type 'number[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'Float[6](https://github.com/20m61/sc4pe_pre2/actions/runs/9973418723/job/27558705325#step:8:7)4Array' is missing the following properties from type 'number[]': pop, push, concat, shift, and 6 more.

  8[7](https://github.com/20m61/sc4pe_pre2/actions/runs/9973418723/job/27558705325#step:8:8) |
  88 |       // FFTデータに基づいて頂点位置と色を更新
> 89 |       const positions = randomMesh.geometry.attributes.position
     |                         ^
  90 |         .array as number[];
  [9](https://github.com/20m61/sc4pe_pre2/actions/runs/9973418723/job/27558705325#step:8:10)1 |       const colors = randomMesh.geometry.attributes.color.array as number[];
  92 |       let needsUpdate = false;
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
0s
```

--- a/src/components/RandomMesh.tsx
+++ b/src/components/RandomMesh.tsx
@@ -1,4 +1,3 @@
-// src/components/RandomMesh.tsx
 import React, { useEffect, useRef } from 'react';
 import * as THREE from 'three';
 
@@ -8,7 +7,7 @@ interface RandomMeshProps {
 
 const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
   const mountRef = useRef<HTMLDivElement>(null);
-  let renderer: THREE.WebGLRenderer;
+  const rendererRef = useRef<THREE.WebGLRenderer | null>(null);
 
   useEffect(() => {
     const mount = mountRef.current;
@@ -24,14 +23,14 @@ const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
     );
 
     // レンダラーの作成または再利用
-    if (!renderer) {
-      renderer = new THREE.WebGLRenderer({ antialias: true });
-      renderer.setSize(window.innerWidth, window.innerHeight);
+    if (!rendererRef.current) {
+      rendererRef.current = new THREE.WebGLRenderer({ antialias: true });
+      rendererRef.current.setSize(window.innerWidth, window.innerHeight);
     } else {
-      renderer.setSize(window.innerWidth, window.innerHeight);
-      renderer.clear();
+      rendererRef.current.setSize(window.innerWidth, window.innerHeight);
+      rendererRef.current.clear();
     }
-    mount.appendChild(renderer.domElement);
+    mount.appendChild(rendererRef.current.domElement);
 
     // ランダムメッシュの作成
     const createRandomMesh = () => {
@@ -77,7 +76,7 @@ const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
     const onWindowResize = () => {
       camera.aspect = window.innerWidth / window.innerHeight;
       camera.updateProjectionMatrix();
-      renderer.setSize(window.innerWidth, window.innerHeight);
+      rendererRef.current?.setSize(window.innerWidth, window.innerHeight);
     };
     window.addEventListener('resize', onWindowResize, false);
 
@@ -87,11 +86,11 @@ const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
 
       // FFTデータに基づいて頂点位置と色を更新
       const positions = Array.from(
-        randomMesh.geometry.attributes.position.array
-      ) as number[];
+        randomMesh.geometry.attributes.position.array as Float32Array
+      );
       const colors = Array.from(
-        randomMesh.geometry.attributes.color.array
-      ) as number[];
+        randomMesh.geometry.attributes.color.array as Float32Array
+      );
       let needsUpdate = false;
 
       for (let i = 0; i < positions.length; i += 3) {
@@ -115,7 +114,7 @@ const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
         randomMesh.geometry.attributes.color.needsUpdate = true;
       }
 
-      renderer.render(scene, camera);
+      rendererRef.current?.render(scene, camera);
     };
     animate();
 
@@ -129,9 +128,9 @@ const RandomMesh: React.FC<RandomMeshProps> = ({ audioData }) => {
       scene.remove(randomMesh);
 
       // レンダラーの破棄
-      renderer.dispose();
+      rendererRef.current?.dispose();
 
-      mount.removeChild(renderer.domElement);
+      mount.removeChild(rendererRef.current!.domElement);
       window.removeEventListener('resize', onWindowResize);
     };
   }, [audioData]);

